controllers/global_search_setup.go

```diff
@@ -0,0 +1,558 @@
+		return err // QUESTION: Should we return here or continue? Should we rollback other changes?
+	}
+
+	// Create configuration resources for each Managed Hub.
+	// MANAGED_HUBS=($(oc get managedcluster -o json | jq -r '.items[] | select(.status.clusterClaims[] |
+	//     .name == "hub.open-cluster-management.io" and .value != "NotInstalled") | .metadata.name'))
+	clusterList, err := r.DynamicClient.Resource(managedClusterResourceGvr).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Error(err, "Failed to list the ManagedClusters to configure global search.")
+		return err // QUESTION: Should we return here or continue? Should we rollback other changes?
+	}
+
+	for _, cluster := range clusterList.Items {
+		isManagedHub := false
+		if cluster.Object["status"] == nil || cluster.Object["status"].(map[string]interface{})["clusterClaims"] == nil {
+			log.V(5).Info("Cluster doesn't have status or clusterClaims.", "cluster", cluster.GetName())
+			continue
+		}
+		clusterClaims := cluster.Object["status"].(map[string]interface{})["clusterClaims"].([]interface{})
+		for _, claim := range clusterClaims {
+			claimMap := claim.(map[string]interface{})
+			if claimMap["name"] == "hub.open-cluster-management.io" && claimMap["value"] != "NotInstalled" {
+				isManagedHub = true
+				break
+			}
+		}
+		if !isManagedHub {
+			log.V(5).Info("Cluster is not a Managed Hub.", "name", cluster.GetName())
+			continue
+		}
+
+		log.V(2).Info("Cluster is a Managed Hub. Configuring global search resources.", "name", cluster.GetName())
+
+		// 3a. Create a ManagedServiceAccount search-global.
+		err = r.createManagedServiceAccount(ctx, cluster.GetName())
+		if err != nil {
+			log.Error(err, "Failed to create ManagedServiceAccount search-global for", "cluster", cluster.GetName())
+			// QUESTION: How should we handle partial errors? Should we continue or rollback?
+		}
+
+		// 3b. Create a ManifestWork search-global-config if it doesn't exist.
+		err = r.createManifestWork(ctx, cluster.GetName())
+		if err != nil {
+			log.Error(err, "Failed to create ManifestWork search-global-config for", "cluster", cluster.GetName())
+			// QUESTION: How should we handle partial errors? Should we continue or rollback?
+		}
+	}
+	log.Info("Global search resources configured.")
+	return err
+}
+
+// Create a ManagedServiceAccount search-global in the Managed Hub namespace.
+// This will create a service account in the Managed Hub and sync the secret containing the access token.
+func (r *SearchReconciler) createManagedServiceAccount(ctx context.Context, cluster string) error {
+	managedSA := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "authentication.open-cluster-management.io/v1beta1",
+			"kind":       "ManagedServiceAccount",
+			"metadata": map[string]interface{}{
+				"name": searchGlobal,
+				"labels": map[string]interface{}{
+					"app": "search",
```

Comment:
As discussed, adding "feature":"globalSearch" to all the new objects we create might be helpful to see all the things we put in to enable this feature.

Reply:
added label `"feature": "global-search"`